File: Leetcode/73 - Set Matrix Zeroes/solution.cpp
void setZeroes(vector<vector<int>> &matrix)
{
    int m = matrix.size();
    int n = matrix[0].size();
    bool setFirstRow = false, setFirstCol = false;
    for (int j = 0; j < n; j++)
    {
        if (matrix[0][j] == 0)
        {
            setFirstRow = true;
            break;
        }
    }
    for (int i = 0; i < m; i++)
    {
        if (matrix[i][0] == 0)
        {
            setFirstCol = true;
            break;
        }
    }
    for (int i = 1; i < m; i++)
    {
        for (int j = 1; j < n; j++)
        {
            if (matrix[i][j] == 0)
            {
                matrix[i][0] = 0;
                matrix[0][j] = 0;
            }
        }
    }
    for (int i = 1; i < m; i++)
    {
        for (int j = 1; j < n; j++)
        {
            if (matrix[i][0] == 0 || matrix[0][j] == 0)
            {
                matrix[i][j] = 0;
            }
        }
    }
    if (setFirstRow)
    {
        for (int j = 0; j < n; j++)
        {
            matrix[0][j] = 0;
        }
    }
    if (setFirstCol)
    {
        for (int i = 0; i < m; i++)
        {
            matrix[i][0] = 0;
        }
    }
}

File: Leetcode/73 - Set Matrix Zeroes/solution.py
def setZeroes(self, matrix: List[List[int]]) -> None:
    m: int = len(matrix)
    n: int = len(matrix[0])

    set_first_row: bool = any(matrix[0][j] == 0 for j in range(n))
    set_first_col: bool = any(matrix[i][0] == 0 for i in range(m))

    # Mark rows and columns
    for i in range(1, m):
        for j in range(1, n):
            if matrix[i][j] == 0:
                matrix[i][0] = 0
                matrix[0][j] = 0

    # Apply markers
    for i in range(1, m):
        for j in range(1, n):
            if matrix[i][0] == 0 or matrix[0][j] == 0:
                matrix[i][j] = 0

    # Zero first row
    if set_first_row:
        for j in range(n):
            matrix[0][j] = 0

    # Zero first column
    if set_first_col:
        for i in range(m):
            matrix[i][0] = 0